src/Compilers/CSharp/Portable/Symbols/Source/SourceUserDefinedOperatorSymbolBase.cs

```diff
@@ -13,7 +12,11 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
     internal abstract class SourceUserDefinedOperatorSymbolBase : SourceMemberMethodSymbol
     {
-        private const TypeCompareKind ComparisonForUserDefinedOperators = TypeCompareKind.IgnoreTupleNames | TypeCompareKind.IgnoreNullableModifiersForReferenceTypes;
+        private const TypeCompareKind ComparisonForUserDefinedOperators =
+            TypeCompareKind.IgnoreTupleNames |
+            TypeCompareKind.IgnoreNullableModifiersForReferenceTypes |
+            TypeCompareKind.IgnoreNativeIntegers;
```

Comment:
From a discussion with @tannergooding, we may want to make native integers equivalent to IntPtr out-of-the-box (without a TypeCompareKind flag). But this can't be done before the "unchecked operators" proposal is implemented.

Reply:
Also requires ldm approval, etc